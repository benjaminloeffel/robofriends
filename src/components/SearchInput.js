import React from "react";
import PropTypes from "prop-types";
/** @jsx jsx */
import { jsx } from "theme-ui";
import { Label, Input } from "@theme-ui/components";

const SearchInput = ({ searchTerm, searchChange }) => {
  return (
    <div>
      <Label htmlFor="search">Search Robots</Label>
      <Input
        id="search"
        sx={{ mb: 3, maxWidth: 250, color: "text" }}
        placeholder="Enter Robot Name"
        value={searchTerm}
        onChange={searchChange}
      />
    </div>
  );
};

SearchInput.propTypes = {
  searchTerm: PropTypes.string.isRequired,
  searchChange: PropTypes.func.isRequired
};

export default SearchInput;
