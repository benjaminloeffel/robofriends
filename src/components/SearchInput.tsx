/** @jsx jsx */
import React from "react";
import { jsx } from "theme-ui";
import { Label, Input } from "@theme-ui/components";
import PropTypes from "prop-types";

type Props = {
  searchTerm: string;
  searchChange(event: React.SyntheticEvent<HTMLInputElement>): void;
};

const SearchInput: React.FC<Props> = ({ searchTerm, searchChange }) => {
  return (
    <div>
      <Label htmlFor="search">Search Robots</Label>
      <Input
        id="search"
        sx={{ mb: 3, maxWidth: 250, color: "text" }}
        placeholder="Enter Robot Name"
        value={searchTerm}
        onChange={searchChange}
      />
    </div>
  );
};

SearchInput.propTypes = {
  searchTerm: PropTypes.string.isRequired,
  searchChange: PropTypes.func.isRequired
};

export default SearchInput;
