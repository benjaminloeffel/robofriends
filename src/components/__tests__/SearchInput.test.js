import React from "react";
import "../../setupTests";
import { shallow } from "enzyme";
import SearchInput from "../SearchInput";

describe("SearchInput", () => {
  const mockProps = {
    searchTerm: "test",
    onSearchChange: jest.fn()
  };

  const wrapper = shallow(<SearchInput {...mockProps} />);

  it("renders the SearchInput component with props", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
