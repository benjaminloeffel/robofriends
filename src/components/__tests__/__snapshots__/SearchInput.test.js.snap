// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders the SearchInput component with props 1`] = `ShallowWrapper {}`;
